webofneeds/won-core/src/main/java/won/protocol/repository/NeedEventContainerRepository.java

```diff
@@ -41,5 +41,14 @@
     @Lock(LockModeType.PESSIMISTIC_WRITE)
     @Query("select c from NeedEventContainer c join MessageEventPlaceholder msg on msg.parentURI = c.parentUri where msg.messageURI = :messageUri")
     public NeedEventContainer findOneByContainedMessageUriForUpdate(@Param("messageUri") URI messageUri);
+    
+    @Query("select case when (count(n) > 0) then true else false end " +
+            "from Need n left outer join Connection con on (n.needURI = con.needURI) "+
+            " where n.needURI = :needUri and "
```

Comment:
that means reading a need is permitted for the need itself and the remote need? but cannot anybody read all need contents?

Reply:
This is used for reading the contents of the need's event container.